CONTRIBUTING.md

```diff
@@ -1,30 +1,151 @@
 
-## Project structure
+## <a name="ps"></a> Project structure
 
 This project is a monorepo managed by [Lerna][lerna-home]. This means that each folder inside the [packages/](../packages) directory represents a package published to npm. The root directory also contains _package.json_ file but this is only used for local development purposes and does not represent something that is ever published to npm (you might notice that it only contains _devDependencies_ and no prod deps 🤷‍♂️).
 
-## Commit messages
+## <a name="cm"></a> Commit messages
 
 All commits you make should adhere to our commit guidelines. We use [conventional commits][conventional-commits] strategy with slight modification of our own - [@lmc-eu/commitlint-config][commitlint-config]. This is later used during release phase to determine how to bump the packages' version numbers based on commit history. 🚀
 
-Message of commit must contain specific phrases, like `<type>(<package>): <commit-message> #<issue-id>` eg.`Feat(web-react): React Modal implementation #DS-331`
+### Commit Message Format
+
+We have very precise rules over how our Git commit messages must be formatted.
+This format leads to **easier to read commit history**.
+
+Each commit message consists of a **header**, a **body**, and a **footer**.
+
+```
+<header>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The `header` is mandatory and must conform to the [Commit Message Header](#commit-header) format.
+
+The `body` is optional but recommended for all commits except for those of type "Docs".
+When the body is present it must be at least 20 characters long and must conform to the [Commit Message Body](#commit-body) format.
+
+The `footer` is optional. The [Commit Message Footer](#commit-footer) format describes what the footer is used for and the structure it must have.
+
+#### <a name="commit-header"></a>Commit Message Header
+
+```
+<type>(<scope>): <short summary>
+  │       │             │
+  │       │             └─⫸ Summary in present tense. Sentence case. No period at the end.
+  │       │
+  │       └─⫸ Commit Scope: design-tokens|icons|web|web-react|web-twig|exporter-css|exporter-svg|
+  │                          ci|repo
+  │
+  └─⫸ Commit Type: Feat|Fix|Perf|Revert|Docs|Style|Refactor|Test|Chore|Deps
+```
+
+The `<type>` and `<summary>` fields are mandatory, the `(<scope>)` field is optional.
+
+##### Type
+
+Must be one of the following:
+
+- **Chore**: Changes to our CI configuration files and scripts (examples: CircleCi, SauceLabs) or changes that affect build system
+- **Deps**: Changes to dependencies
+- **Docs**: Documentation only changes
+- **Feat**: A new feature
+- **Fix**: A bug fix
+- **Perf**: A code change that improves performance
+- **Refactor**: A code change that neither fixes a bug nor adds a feature
+- **Revert**: Reverting code changes
+- **Test**: Adding missing tests or correcting existing tests
+- **Style**: Styling of components of changes to code style
```

Comment:
This is unclear to me… Would a shorter version just do?

```suggestion
- **Style**: Changes to code style
```

Reply:
Fixed.